doc/source/developing/creating_frontend.rst

```diff
@@ -317,3 +317,21 @@ And that just about covers it. Please feel free to email
 `yt-users <https://mail.python.org/archives/list/yt-users@python.org/>`_ or
 `yt-dev <https://mail.python.org/archives/list/yt-dev@python.org/>`_ with
 any questions, or to let us know you're thinking about adding a new code to yt.
+
+
+How to add extra dependencies ?
+-------------------------------
+
+It is required that a specific target be added to ``pyproject.toml`` to define a list
+of additional requirements (even if empty), see :ref:`install-additional`.
+
+At runtime, extra third party dependencies should be loaded lazily, meaning their import
+needs to be delayed until actually needed. This is achieved by importing a wrapper from
+``yt.utitilies.on_demand_imports.py``, instead of the actual package like so
+
+.. code-block:: python
+
+    from yt.utilities.on_demand_imports import _mypackage as mypackage
+
+Such import statements can live at the top of a module without generating overhead or errors
+in case the actual package isn't installed.
```

Comment:
```suggestion
in case the actual package isn't installed.

If the extra third party dependency is new, a new import wrapper must also be added. To do so, 
follow the example of the existing wrappers in ``yt.utilities.on_demand_imports.py``.
```

Maybe this sentence should go earlier, but I thought it was worth explicitly mentioning (even though it **should** be obvious that you can't import `_mypackage` if it doesn't exist yet)... 

I think the existing wrappers should be clear enough for someone to use as examples, but if you felt like it, it wouldn't hurt to explicitly document how to write a new on demand wrapper (inherit from `OnDemand`, add `safe_import` functions for any submodules that you will need, etc.).

Reply:
I think your suggestion covers it satisfyingly, and has the great advantage of not easily getting out-of-sync with the code !